### PR TITLE
Updates the order of TonY Portal records and adds UI for user sorting/filtering

### PR DESCRIPTION
**Pull request by StanfordMCP**
This fix updates the TonY Portal to sort records chronologically. This update also adds a Datatable framework to provide users the ability to sort records by column and to filter records based on individual keywords.

### CODE REVIEW

File: tony-portal/app/views/tableMetadata.scala.html [hunk 1]
```diff
@@ -1,34 +1,68 @@
...
   }
-</table>
\ No newline at end of file
+  </tbody>
+</table>
+</body>
+
+
+  <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
+  <link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.10.19/css/jquery.dataTables.css">
```
Comment (oliverhu) [inline tony-portal/app/views/tableMetadata.scala.html]: let's download all these resources into resource folder

**Comment by erwa** [comment]
@StanfordMCP , can you upload a screenshot of what the new UI looks like with sorting and filtering capabilities?

**Comment by StanfordMCP** [comment]
@erwa This is a screenshot of the updated UI, in this version (not yet committed) I added footer filters to search by column. 

image

**Comment by erwa** [comment]
Very cool! Love the new UI.

**Comment by oliverhu** [comment]
@erwa looks like you are just wfh 😄

**Comment by erwa** [comment]
Just checking in on how our fish TonY is doing :). You need to feed it daily and clean its water tank from time to time.

File: tony-portal/app/views/tableMetadata.scala.html (hunk 1 above)
Comment (StanfordMCP) [reply tony-portal/app/views/tableMetadata.scala.html]: @oliverhu Addressed comment, please take a look.

File: tony-portal/app/views/tableMetadata.scala.html [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+<html>
+
+  <head>
+
     <!-- Required meta tags for Bootstrap-->
-  <meta charset="utf-8">
-  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-    <!-- Bootstrap CSS -->
-  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css" integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS" crossorigin="anonymous">
+  <meta charset="utf-8"/>
```
Comment (erwa) [inline tony-portal/app/views/tableMetadata.scala.html]: Nit: I think this should be indented two more spaces

File: tony-portal/app/views/tableMetadata.scala.html [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
+        color: rgba(0, 0, 0, 0)
+      }
 
+      body {
+        margin:12px;
+      }
+
+  </style>
+
+</head>
```
Comment (erwa) [inline tony-portal/app/views/tableMetadata.scala.html]: Should the <style> section and the </head> tag all be indented two more spaces?

File: tony-portal/app/views/tableMetadata.scala.html [hunk 4]
```diff
@@ -44,25 +63,59 @@
...
   </tbody>
+    <tfoot>
+      <tr>
+        <th>App ID</th>
+        <th>Configuration</th>
+        <th>Started</th>
+        <th>Completed</th>
+        <th>User</th>
+        <th>Status</th>
+        <th class = "rmv_elmt">ResourceManager Link</th>
```
Comment (erwa) [inline tony-portal/app/views/tableMetadata.scala.html]: Can you add a comment explaining what "rmv_elmt" is for?

File: tony-portal/app/views/tableMetadata.scala.html [hunk 5]
```diff
@@ -44,25 +63,59 @@
...
+          var title = $('#table_id tfoot th').eq( $(this).index() ).text();
+          if (title !="ResourceManager Link")
+            $(this).html( '<input type="text" placeholder="Search '+title+'" data-index="'+i+'" />' );
+        } );
+
+        // DataTable (uncomment lines to add scrolling)
+        var table = $('#table_id').DataTable( {
+          // scrollY:        "300px",
+          // scrollX:        true,
+          // scrollCollapse: true,
```
Comment (erwa) [inline tony-portal/app/views/tableMetadata.scala.html]: I would just remove these commented lines and the comment on line 100.

File: tony-portal/app/views/tableMetadata.scala.html (hunk 2 above)
Comment (StanfordMCP) [reply tony-portal/app/views/tableMetadata.scala.html]: @erwa reformatted code, please take a look

File: tony-portal/app/views/tableMetadata.scala.html (hunk 3 above)
Comment (StanfordMCP) [reply tony-portal/app/views/tableMetadata.scala.html]: @erwa reformatted code, please take a look

File: tony-portal/app/views/tableMetadata.scala.html (hunk 4 above)
Comment (StanfordMCP) [reply tony-portal/app/views/tableMetadata.scala.html]: @erwa added comment, please take a look

File: tony-portal/app/views/tableMetadata.scala.html (hunk 5 above)
Comment (StanfordMCP) [reply tony-portal/app/views/tableMetadata.scala.html]: @erwa addressed issue, please take a look

**Comment by StanfordMCP** [comment]
@erwa removed extraneous non-min files

File: tony-portal/app/views/tableMetadata.scala.html [hunk 6]
```diff
@@ -10,25 +42,84 @@
...
+    <td>
+      <a href=@s.getRMLink target="_blank">Link</a>
+    </td>
+  </tr>
   }
-</table>
\ No newline at end of file
+  </tbody>
+
+@*  rmv_elmt tags the labels that will not have a footer filter*@
```
Comment (erwa) [inline tony-portal/app/views/tableMetadata.scala.html]: Cool, didn't know there was special syntax for server-side comments.

**Comment by erwa** [comment]
In https://travis-ci.org/linkedin/TonY/builds/562851319?utm_source=github_status&utm_medium=notification, BrowserTest (line 4242) failed with

```
        Caused by:
        com.gargoylesoftware.htmlunit.ScriptException: ReferenceError: "$" is not defined. (script in http://localhost:19001/ from (106, 32) to (130, 10)#108)
```

@StanfordMCP , can you take a look and fix this?

**Comment by StanfordMCP** [comment]
@erwa added fixes, please take a look.

**Comment by StanfordMCP** [comment]
@UWFrankGu addressed comment, please check.

**Comment by erwa** [comment]
Nice work! Also, thanks for fixing the build!

**Pull request merged**